qiskit/result/result.py

```diff
@@ -25,47 +26,135 @@
 from qiskit.qobj.utils import MeasLevel
 from qiskit.qobj import QobjHeader
 
+_MISSING = object()
+
 
 class Result:
     """Model for Results.
 
+    .. deprecated:: 1.4
+        The use of positional arguments in the constructor of :class:`.Result`
+        is deprecated as of Qiskit 1.4, and will be disabled in Qiskit 2.0.
+        Please set all arguments using kwarg syntax, i.e: ``Result(backend_name="name", ....)``.
+        In addition to this, the ``qobj_id`` argument is deprecated and will no longer
+        be used in Qiskit 2.0. It will, however, still be possible to set ``qobj_id`` as a
+        generic kwarg, which will land in the metadata field with the other generic kwargs.
+
     Attributes:
-        backend_name (str): backend name.
-        backend_version (str): backend version, in the form X.Y.Z.
-        qobj_id (str): user-generated Qobj id.
-        job_id (str): unique execution id from the backend.
-        success (bool): True if complete input qobj executed correctly. (Implies
+        backend_name (str): (REQUIRED) backend name.
```

Comment:
I'm wondering if we want this under `Args` instead of attributes? They're both really because the constructor just does `self.backend_name = backend_name`. But right now this is under attributes.

Reply:
Yes, args would help covey the message that... well, these are the args. Will do.

Reply:
done [b9d9fee](https://github.com/Qiskit/qiskit/pull/13868/commits/b9d9fee0472faf519a4b1306825e44194161050a)